### PR TITLE
Make dev environment windows friendly

### PR DESCRIPTION
**Pull request by increddibelly**
change the container setup to make it work on Windows as well

### CODE REVIEW

**Comment by increddibelly** [comment]
requirements for windows platform:
- git bash ( comes with GitExtensions )
- [gsudo](https://github.com/gerardog/gsudo) package (adds sudo command)
- todo: find replacement for lsof

**Comment by noud-github** [comment]
the function:
```
function check_hosts_newline () {
    hosts_lastchar=$(tail -c 1 $hostfile)
    [[ "$hosts_lastchar" != "" ]] && echo '' | sudo tee -a $hostfile
} 

```
is broken as $hostfile is not defined here.

**Comment by noud-github** [comment]
I think it is better to make $hostfile a script wide variable that fixes this.

**Comment by Petervw** [comment]
> requirements for windows platform:
> 
> * git bash ( comes with GitExtensions )
> * [gsudo](https://github.com/gerardog/gsudo) package (adds sudo command)
> * todo: find replacement for lsof

Can you also define these requirements in the Readme file of the project?

**Comment by increddibelly** [comment]
made some changes;
- global hostfile variable
- look for windows first; bash shell application creates /etc/hosts so that exists on both platforms but C;/Windows/System32 does not.

**Review by noud-github — APPROVED** [review]
looks good,

**Review by Petervw — COMMENTED** [review]
Can you also test those changes on a macbook before merging? I want to be sure everything is still working correctly.

File: make.sh [hunk 1]
```diff
@@ -28,23 +30,40 @@ kill_port_80_usage () {
     fi
 }
 
+find_hostfile () {
+    # try windows first, mac second; add future platforms here.
+    # make sure there aren't any conflicts e.g. the windows bash host may also provide an /etc/hosts file
+    for hosts_candidate in "C:/windows/system32/drivers/etc/hosts" "/etc/hosts"; do
+        echo -n "looking for hostfile at "$hosts_candidate
```
Comment (Djennez) [inline make.sh:37]:
```suggestion
        echo "Looking for hostfile at ${hosts_candidate}"
```
No newline here will only clutter the output.
Capitalized sentence
Keeping the variable in the echo quotes for consistency

File: make.sh [hunk 2]
```diff
@@ -28,23 +30,40 @@ kill_port_80_usage () {
...
+        if [ -f "$hosts_candidate" ]; then
+            #file exists, assign hostfile in outer function to current value
+            echo -n "Found host file at ${hosts_candidate}"
```
Comment (Djennez) [inline make.sh:40]:
```suggestion
            echo "Found host file at ${hosts_candidate}" 
```
Removed no newline for same reason: keep output clean.

**Review by noud-github — CHANGES_REQUESTED** [review]
ther changed so mutch tat my aproval is not valit any more

File: make.sh [hunk 3]
```diff
@@ -28,23 +36,46 @@ kill_port_80_usage () {
...
+            [[ "$index" == 0 ]] && $platform = APPLE
+            [[ "$index" == 1 ]] && $platform = WINDOWS
+
+            return;
+        fi
+        index = 1 + $index
+    done
+
+    echo "host file not found!"
+    exit 404
```
Comment (Djennez) [inline make.sh:59]:
```suggestion
    exit 1
```
With no special meaning, or usage, just use 1 as the exit code to indicate an error. Error codes greater than an 8-bit integer (255) can cause unwanted behavior.

File: start.sh [hunk 4]
```diff
@@ -111,20 +180,18 @@ echo "Outputting logs now:"
 docker-compose logs -f &
 PROCESS=$!
 
+#watchdog
 while [ "$STOPPING" != 'true' ]; do
```
Comment (noud-github) [inline start.sh:184]: This is a workaround for a MAC issue: a system sleep is not communicated to the `docker VM` this screws up the clock in the docker containers, making it dead slow, does this also exits in windows? probably no need to monitor the container clock in windows.

**Review by noud-github — COMMENTED** [review]
the start.sh now has functions and command alternating can we move the function together that will add to the readability.

File: start.sh (hunk 4 above)
Comment (increddibelly) [reply start.sh:184]: I need a bit more info - let's do a call.

**Comment by increddibelly** [comment]
- [ ] test on windows
- [ ] test on mac

**Comment by increddibelly** [comment]
retested on windows after 1e019f8

**Pull request merged**